### PR TITLE
Add hasCopyConstructor and hasPostblit traits

### PR DESCRIPTION
**Pull request by adamdruppe**
I forked @edi33416 's PR here https://github.com/dlang/dmd/pull/10265 and split it into two pieces as mentioned in the comments.

### CODE REVIEW

File: src/dmd/traits.d [hunk 1]
```diff
@@ -627,6 +629,49 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         }
         return True();
     }
+    if (e.ident == Id.hasPostblit)
```
Comment (clYd3r) [inline src/dmd/traits.d]: The two traits share most of their code. So here

```d
if (e.ident == Id.hasPostblit || e.ident == Id.hasCopyConstructor)
```

File: src/dmd/traits.d [hunk 2]
```diff
@@ -627,6 +629,49 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
...
+        {
+            e.error("type expected as second argument of __traits `%s` instead of `%s`",
+                e.ident.toChars(), o.toChars());
+            return new ErrorExp();
+        }
+
+        Type tb = t.baseElemOf();
+        if (auto sd = tb.ty == Tstruct ? (cast(TypeStruct)tb).sym : null)
+        {
+            return sd.postblit ? True() : False();
```
Comment (clYd3r) [inline src/dmd/traits.d]: and then here


```d
 return (e.ident == Id.hasPostblit) ? (sd.postblit ? True() : False())
                                    : (sd.hasCopyCtor ? True() : False());
```

**Review by ghost — COMMENTED** [review]
The two traits can be written together with a small change

File: src/dmd/traits.d (hunk 2 above)
Comment (adamdruppe) [reply src/dmd/traits.d]: yeah i originally literally just mindlessly copy/pasted it but this is nicer, change made just waiting on the autotester to make sure i didnt mess up

**Review by WalterBright — APPROVED** [review]

**Comment by wilzbach** [comment]
Please don't forget the dlang.org update.

**Pull request merged**